cmd/warpforge/watch.go

```diff
@@ -23,20 +32,132 @@ var watchCmdDef = cli.Command{
 	Action: util.ChainCmdMiddleware(cmdWatch,
 		util.CmdMiddlewareLogging,
 		util.CmdMiddlewareTracingConfig,
+		util.CmdMiddlewareCancelOnInterrupt,
 	),
+	Flags: []cli.Flag{
+		&cli.BoolFlag{
+			Name:  "socket",
+			Usage: "Experimental flag for getting execution status of the watched plot externally via unix socket",
+		},
+	},
 }
 
-func cmdWatch(c *cli.Context) error {
-	if c.Args().Len() != 1 {
-		return fmt.Errorf("invalid args")
+type watchCfg struct {
+	path         string
+	recursive    bool
+	socket       bool
+	plotCfg      wfapi.PlotExecConfig
+	recentStatus int
+}
+
+func (c *watchCfg) handle(ctx context.Context, conn net.Conn) error {
+	// in lieu of doing anything complicated, shoving a status int down the pipe is sufficient
+	// for the unix socket implementation we can now use netcat or similar for a status.
+	// I.E. nc -U ./sock
+	// OR socat - UNIX-CONNECT:./sock
+	defer conn.Close()
+	enc := json.NewEncoder(conn)
+	err := enc.Encode(c.recentStatus)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *watchCfg) serveLoop(ctx context.Context, l net.Listener) error {
+	log := logging.Ctx(ctx)
+	for {
+		conn, err := l.Accept() // blocks, doesn't accept a context.
+		if err != nil {
+			return err
+		}
+		go func() {
```

Comment:
I think it may be a good idea to habituate having some small struct for a task before hitting the `go` keyword.

This most definitely works, but I worry that it rapidly becomes harder to reason about, name the thing, and so forth.

Some functions are _so_ small that an anonymous func handed to the `go` keyword without further aplomb is fine (the `go CancelOnInterrupt` is an example I'd put on this side), but things that are bigger, or that we spawn many of, seem worth giving a bit heavier treatment.

In this particular case, I think there's a whole "actor" worth of code here and a struct to organize it will help.

We don't necessarily have to do it in this pass, but even if we choose to punt in this PR, I strongly suspect we'll extract this and introduce such structures in a second pass.

Reply:
(I might go play with supervisor frameworks for a bit today and come back with more specific opinions.  I won't block us on that, though.)